surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/TestNGReporter.java

```diff
@@ -183,4 +183,21 @@ public void onConfigurationSuccess( ITestResult result )
         //onTestSuccess( result );
     }
 
+    /**
+     * Acquire a better representation of the test name that includes parameters and the invocation count, if there are
+     * any parameters
+     *
+     * @param result the test result to extract from
+     * @return a descriptive name for the test
+     */
+    private static String testName( ITestResult result )
+    {
+        Object[] parameters = result.getParameters();
+        if ( parameters == null || parameters.length == 0 )
+        {
+            return result.getName();
+        }
+        return result.getName() + Arrays.toString(
```

Comment:
Here are the values of `parameters`. It may be unreadable with complext `toString() value` on the HTML report. In JUnit5 we do not put values but the types (simple class names). Anyway, the Java is strictly static and so the types identify the unique method signature.

Reply:
I used the values here on purpose, rather than the types: Yes, there may be readability problems for complex types, but the whole point was that you could quickly see at a glance *for which exact values* the parametrized test has failed. I don't really see any additional value in adding just the parameter types; I think most people won't have overloaded test methods that do different things with regards to their parameters. TestNG's own HTML report also contains these, no matter if my fix is included or not:

![testng-report](https://user-images.githubusercontent.com/1269520/78109774-31a2ea00-73fa-11ea-974e-59f62c497ab1.png)

Reply:
ok, the users have to write brief and readable value returned by `toString()`.
Can you pls write the integration test now?

Reply:
Alright, going to be working on that at some point next week, depending on how much else is happening work-wise. Should probably be done by friday.

Reply:
I hope you did not forget us ;-)